packages/frontend/src/renderer/components/CreateJoinCommunity/JoinCommunity/JoinCommunity.test.tsx

```diff
@@ -0,0 +1,326 @@
+
+    userEvent.type(input, url)
+    userEvent.click(button)
+
+    await waitFor(() => expect(handleCommunityAction).not.toBeCalled())
+
+    const message = await screen.findByText(error)
+    expect(message).toBeVisible()
+  })
+
+  it('blocks submit button if connection is not ready', async () => {
+    const handleCommunityAction = jest.fn()
+
+    const component = <PerformCommunityActionComponent
+      open={true}
+      handleClose={() => {}}
+      communityAction={CommunityAction.Join}
+      handleCommunityAction={handleCommunityAction}
+      handleRedirection={() => { }}
+      isConnectionReady={false}
+      community={false}
+    />
+
+    const result = renderComponent(component)
+
+    const textInput = result.queryByPlaceholderText(inviteLinkField().fieldProps.placeholder)
+    expect(textInput).not.toBeNull()
+
+    userEvent.type(textInput, 'nqnw4kc4c77fb47lk52m5l57h4tcxceo7ymxekfn7yh5m66t4jv2olad')
+
+    const submitButton = result.queryByRole('button')
+    expect(submitButton).not.toBeNull()
+    expect(submitButton).toBeDisabled()
+
+    expect(handleCommunityAction).not.toBeCalled()
+  })
+
+  it('handles redirection if user clicks on the link', async () => {
+    const handleRedirection = jest.fn()
+
+    const component = <PerformCommunityActionComponent
+      open={true}
+      handleClose={() => {}}
+      communityAction={CommunityAction.Join}
+      handleCommunityAction={() => {}}
+      handleRedirection={handleRedirection}
+      isConnectionReady={true}
+      community={false}
+    />
+
+    const result = renderComponent(component)
+
+    const switchLink = result.queryByText('create a new community')
+    expect(switchLink).not.toBeNull()
+
+    userEvent.click(switchLink)
+
+    expect(handleRedirection).toBeCalled()
+  })
+
+  // TODO: move this scenario to rtl-tests and provide socket (nectar sagas) to prepareStore method
+  it.skip('user rejoins to remembered community without user data', async () => {
+    const { store } = await prepareStore({
+      [StoreKeys.Socket]: {
+        ...new SocketState(),
+        isConnected: true
+      },
+      [StoreKeys.Modals]: {
+        ...new ModalsInitialState(),
+        [ModalName.joinCommunityModal]: { open: true }
+      },
+      [NectarStoreKeys.Communities]: {
+        ...new communities.State()
+      }
+    })
+
+    const factory = await getFactory(store)
+
+    await factory.create<
+    ReturnType<typeof communities.actions.addNewCommunity>['payload']
+    >('Community')
+
+    renderComponent(
+      <>
+        <JoinCommunity />
+        <CreateUsername />
+      </>,
+      store
+    )
+
+    const createUsernameTitle = screen.getByText('Register a username')
+    expect(createUsernameTitle).toBeVisible()
+  })
+
+  // TODO: move this scenario to rtl-tests and provide socket (nectar sagas) to prepareStore method
+  it.skip('user rejoins to remembered community with user certificate', async () => {
```

Comment:
Should we do this now?

Reply:
That's not related to this task at all. I've already told Damian that those test should be changed